UnityProject/Assets/Scripts/Systems/StationObjectives/StationObjectiveManager.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Text;
+using System.Linq;
+using DiscordWebhook;
+using DatabaseAPI;
+using Messages.Server.LocalGuiMessages;
+using Strings;
+using Managers;
+using Random = UnityEngine.Random;
+
+namespace StationObjectives
+{
+	public class StationObjectiveManager : MonoBehaviour
+	{
+		public GameManager gameManager;
+
+		[SerializeField]
+		[Tooltip("Stores all station objective data.")]
+		private StationObjectiveData stationObjectiveData = null;
+
+		public static StationObjectiveManager Instance;
+
+		[NonSerialized] public StationObjective activeObjective;
+
+		private List<Vector2> asteroidLocations = new List<Vector2>();
+
+		private void Awake()
+		{
+			if (Instance == null)
+			{
+				Instance = this;
+			}
+			else
+			{
+				Destroy(gameObject);
+			}
+		}
+		void OnEnable()
+		{
+			EventManager.AddHandler(EVENT.RoundEnded, ResetObjectives);
+		}
+		void OnDisable()
+		{
+			EventManager.RemoveHandler(EVENT.RoundEnded, ResetObjectives);
+		}
+
+		//public ;
+
+		public void RemindStation()
+		{
+			GameManager.Instance.CentComm.MakeCommandReport(StationObjectiveReport(),CentComm.UpdateSound.Notice);
+		}
+		private string StationObjectiveReport()
+		{
+			var report = new StringBuilder(activeObjective.Description);
+			report.Replace("STATIONNAME", MatrixManager.MainStationMatrix.GameObject.scene.name);
+
+			foreach (var location in asteroidLocations)
+			{
+				report.AppendFormat(" <size=24>{0}</size> ", Vector2Int.RoundToInt(location));
+			}
+
+			return report.ToString();
+		}
+
+		public void ServerChooseObjective()
+		{
+			foreach (var body in gameManager.SpaceBodies)
+			{
+				if (body.TryGetComponent<Asteroid>(out _))
+				{
+					asteroidLocations.Add(body.ServerState.Position);
+				}
+			}
+
+			int randomPosCount = Random.Range(1, 5);
+			for (int i = 0; i <= randomPosCount; i++)
+			{
+				asteroidLocations.Add(gameManager.RandomPositionInSolarSystem());
+			}
+			asteroidLocations = asteroidLocations.OrderBy(x => Random.value).ToList();
+
+			activeObjective = stationObjectiveData.GetRandomObjective();
+			GameManager.Instance.CentComm.MakeQuietCommandReport(StationObjectiveReport());
+		}
+
+		public void ShowStationStatusReport()
+		{
+			StringBuilder statusSB = new StringBuilder($"<color=white><size=60><b>End of Round Report</b></size></color>\n\n", 200);
+
+			var message = $"End of Round Report on {ServerData.ServerConfig.ServerName}\n";
```

Comment:
use StringBuilder for the entire string, don't mix with $string

Reply:
what does that mean

Reply:
you don't need to create a new variable ``message`` just append to statusSB. Also don't use $"{var} string" or +=  with StringBuilder, that's what StringBuilder is for,

Reply:
the message part is for discord integration, gilles. it's not part of the stringbuilder at all.

Reply:
using both StringBuilder and string interpolation is confusing. Why you need the two of them when they have seemingly the same content?

Reply:
mostly just different syntaxes